packages/playground/src/components/execution/key-value-table.tsx

```diff
@@ -25,13 +25,98 @@ interface KeyValueTableProps {
   tableSpec: KeyValuePair[]
 }
 
+//manage style for repeated use
+const accordionContentStyle = 'w-full pl-0 pr-0 border-0 pb-0'
+const specTitleStyle = 'text-ring flex-grow text-left -ml-1'
+
 export const KeyValueTable: React.FC<KeyValueTableProps> = ({
   className,
   tableTitleName,
   tableTitleVal,
   tableSpec
 }) => {
-  // ToDO:const getAccordian = () => <div></div> // do this function recursively
+  const renderListItems = (items: KeyValuePair[], level: number = 1) => {
+    //detect if the listItems is objects or array, tailwind css will not generate
+    const listItems = Array.isArray(items) ? items : [items]
+
+    return listItems.map((item, index: number) => {
+      if (typeof item.value === 'string') {
+        return (
+          <ul className="border-b" key={index}>
+            <li className="py-2.5 inline-block w-1/2 " style={{ paddingLeft: `${level * 1 + 2}rem` }}>
+              <Text size={2} weight="normal" className="text-ring">
+                {item.name}
+              </Text>
+            </li>
+            <li className="pr-2.5 py-2.5 inline-block w-1/2">
+              <Text size={2} weight="normal" className="text-ring">
+                {item.value}
+              </Text>
+            </li>
+          </ul>
+        )
+      } else if (Array.isArray(item.value) || typeof item.value === 'object') {
+        return (
+          <Accordion type="single" key={index} className="border-0" defaultValue={item.name} collapsible>
+            <AccordionItem value={item.name} className="border-0">
+              <AccordionTrigger
+                className="w-full pt-2 pb-2 pr-4 flex"
+                leftChevron
+                style={{ paddingLeft: `${level * 1 + 2}rem` }}>
+                <Text size={2} weight="normal" className={specTitleStyle}>
```

Comment:
Overriding the Text component with className here, but alignment is already a Text prop. Do we really need to diverge with all the styling on this element so much?

Reply:
Just test for a bit for accordion text in the accordion trigger but it's not intuitive handle for accordion text even I over-written the text-align: left on its className. For the further maintaining, I used<Text> first. I've written on this branch for fixing some style and table cell, please wait for merging this branch if you would like to fix anything related on this page. Thanks!
<img width="1248" alt="Screenshot 2024-09-03 at 5 37 33 PM" src="https://github.com/user-attachments/assets/7206e126-a999-4249-8d2b-17e283e26af2">

<img width="1724" alt="Screenshot 2024-09-03 at 5 35 17 PM" src="https://github.com/user-attachments/assets/0f8c1c91-582e-4640-b3df-8564c07bf936">